### PR TITLE
Add CI/CD docs for MI

### PR DESCRIPTION
**Pull request by arunans23**
Resolves https://github.com/wso2/micro-integrator/issues/1241

### CODE REVIEW

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-overview.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Building a Jenkins CI/CD Pipeline for Micro Integrator
+
+## Overview
+
+This is a guide of a reference implementation on setting up a CI/CD process for Micro Integrator. This guide contains two parts.
```
Comment (samgnaniah) [inline en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-overview.md:5]: Can we add some information on when someone would do this or in what exact scenario? It's better to have that kind of content in the overview.

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-overview.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+1. VM based CI/CD
```
Comment (samgnaniah) [inline en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-overview.md:6]: Let's add links to the subtopics.

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-overview.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+2. Kubernetes based CI/CD 
+
+## Phases of SDLC
+
+Considering a deployment with having three typical environments as follows:
```
Comment (samgnaniah) [inline en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-overview.md:11]:
```suggestion
Let's consider a deployment that has three typical environments as follows:
```

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-overview.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+*   Data Source Configs module
+*   Data Service Configs module
+*   Registry Resource module
+*   Mediator module
+*   Composite Exporter module
+*   Connector Exporter module
+*   Docker Exporter module
+*   Kubernetes Exporter module
+
+Please note that you are not allowed to create Nested Integration projects. Also for CI/CD, it is recommended to maintain one git repo per Integration project.
```
Comment (samgnaniah) [inline en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-overview.md:31]:
```suggestion
Please note that you are not allowed to create Nested Integration projects. Also, for CI/CD, it is recommended to maintain one GitHub repository per Integration project.
```

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-vm.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
+[![VM based CI/CD for Micro Integrator]({{base_path}}/assets/img/deploy/mi-cicd-VM.png)]({{base_path}}/assets/img/deploy/mi-cicd-VM.png)
+
+#### Overview
+
+There are two kinds of jobs in Jenkins. One kind is for the Integration projects. Another kind is for the Descriptor repositories. 
+
+*   We need to maintain one jenkins job per Integration Project repo.
```
Comment (samgnaniah) [inline en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-vm.md:7]:
```suggestion
*   We need to maintain one Jenkins job per Integration Project repository.
```

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-vm.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+*   The Integration Project has to be a Maven Multi Module project and it has to contain one Composite Exporter module. By default, the script supports one Exporter module. Users can customize it to support multiple Exporter modules per Integration Project.
```
Comment (samgnaniah) [inline en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-vm.md:8]: Shall we link to the description of MMM project?

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-vm.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+*   The build phase of the job will build the Integration project and run the unit tests if a Unit test server is configured.
+*   The release phase of the job will publish the CApps to the nexus repository and create a release tag in github.
```
Comment (samgnaniah) [inline en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-vm.md:10]:
```suggestion
*   The release phase of the job will publish the CApps to the Nexus repository and create a release tag in GitHub.
```

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-vm.md (hunk 5 above)
Comment (samgnaniah) [reply en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-vm.md:7]: Can we change "jenkins" to "Jenkins" everywhere?

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-overview.md (hunk 1 above)
Comment (arunans23) [reply en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-overview.md]: Added some info. Please review

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-overview.md (hunk 2 above)
Comment (arunans23) [reply en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-overview.md]: Done

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-vm.md (hunk 6 above)
Comment (arunans23) [reply en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-vm.md]: Done

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-k8s.md [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+[![K8s based CI/CD for Micro Integrator]({{base_path}}/assets/img/deploy/mi-cicd-K8s.png)]({{base_path}}/assets/img/deploy/mi-cicd-K8s.png)
+
+#### Overview
+
+There are two kinds of jobs in Jenkins that we need to maintain. One for the Integration project repositories. Another kind is for the Deployment Descriptor repositories. 
+
+#### Integration Project Build Job
+- We need to maintain one Jenkins job per Integration Project repository.
+- The Integration Project has to be a [Maven Multi Module project]({{base_path}}/integrate/develop/create-integration-project/#maven-multi-module-projects) and it has to contain one Kubernetes Exporter module. 
+- The build phase of the job will build the Integration project and run the unit tests if a Unit test server, if configured.
```
Comment (samgnaniah) [inline en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-k8s.md:12]:
```suggestion
- The build phase of the job will build the Integration project and run the unit tests if a Unit test server is configured.
```

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-k8s.md [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+- The release phase of the job generates docker images using the provided repository, name and project version and pushes the image to the configured docker registry and creates a release tag in github.
```
Comment (samgnaniah) [inline en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-k8s.md:13]:
```suggestion
- The release phase of the job generates Docker images using the provided repository, name, and project version and pushes the image to the configured Docker registry and creates a release tag in GitHub.
```
Please change "docker" to "Docker" and "github" or other variants to "GitHub" in all places.

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-k8s.md [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+
+#### Deployment Descriptor Build Job
+- We need to maintain one Jenkins job per Environment.
+- There will be descriptor files for each project inside a separate folder. These will be generated in Integration Studio when you create a Kubernetes Exporter Module.
+- To use a new version / rollback to a previous version, the user should define the version inside the respective descriptor file (integration_cr.yaml or integration_k8s.yaml) and commit to the respective branch.
```
Comment (samgnaniah) [inline en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-k8s.md:18]:
```suggestion
- To use a new version / rollback to a previous version, the user should define the version inside the respective descriptor file `integration_cr.yaml` or `integration_k8s.yaml` and commit to the respective branch.
```

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-k8s.md [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+- This job contains only the build phase. This will apply the relevant yaml files in the relevant environment.
+
+### Kubernetes artifacts
+Integration Studio will be used to generate kubernetes artifacts while creating Kubernetes exporter module. There are two kinds of Kubernetes artifacts that you can create in Integration Studio.
```
Comment (samgnaniah) [inline en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-k8s.md:22]:
```suggestion
Integration Studio will be used to generate Kubernetes artifacts, while creating the Kubernetes exporter module. There are two kinds of Kubernetes artifacts that you can create in Integration Studio. 
```

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-k8s.md [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+
+### Kubernetes artifacts
+Integration Studio will be used to generate kubernetes artifacts while creating Kubernetes exporter module. There are two kinds of Kubernetes artifacts that you can create in Integration Studio. 
+These will be generated inside a folder with the Project Name.
+
+*   Pure Kubernetes artifacts 
+    *   Deployment.yaml
+    *   Service.yaml
+    *   ConfigMap.yaml
+*   K8s-Operator based kubernetes artifact
```
Comment (samgnaniah) [inline en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-k8s.md:29]:
```suggestion
*   K8s-Operator based Kubernetes artifact 
```

File: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-k8s.md (hunk 9 above)
Comment (arunans23) [reply en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-k8s.md]: Done

**Review by samgnaniah — APPROVED** [review]

**Pull request merged**